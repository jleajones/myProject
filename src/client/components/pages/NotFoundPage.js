import React from 'react';
import { Helmet } from 'react-helmet';

const NotFoundPage = ({ staticContext }) => {
  if (staticContext) {
    staticContext.statusCode = 404;
  }
  return (
      <div>
        <Helmet>
          <title>Not Found</title>
          <meta name="description" content="This is the About..." />
        </Helmet>
        <div>404 Page</div>
      </div>
  );
};

export default NotFoundPage;
