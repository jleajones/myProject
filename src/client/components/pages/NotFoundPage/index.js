import React from 'react';
import { Helmet } from 'react-helmet';

import Page from '@shared/Page';
import PropTypes from 'prop-types';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

/**
 *
 * @param staticContext
 * @returns {*}
 * @constructor
 */
const NotFoundPage = ({ staticContext }) => {
  if (staticContext) {
    staticContext.statusCode = 404;
  }
  return (
    <Page pageName="NotFound">
      <Helmet>
        <title>404: Page Not Found</title>
        <meta name="description" content="This is the About..." />
      </Helmet>

      <Container>
        <Grid container spacing={2}>
          <Grid item>
            <Typography variant="h6">Whoops...</Typography>
            <Typography>
              Something went wrong, please check your url and try again!
            </Typography>
          </Grid>
        </Grid>
      </Container>
    </Page>
  );
};

NotFoundPage.defaultProps = {
  staticContext: {}
};

NotFoundPage.propTypes = {
  staticContext: PropTypes.shape({
    statusCode: PropTypes.number
  })
};

export default NotFoundPage;
