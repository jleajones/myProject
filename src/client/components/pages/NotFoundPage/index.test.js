import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '.';

describe('<NotFoundPage />', () => {
  test('Renders component without crashing', () => {
    const component = shallow(<NotFoundPage />);

    expect(component).toMatchSnapshot();
  });
});
