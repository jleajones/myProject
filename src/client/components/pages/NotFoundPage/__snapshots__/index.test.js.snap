// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> Renders component without crashing 1`] = `
<Connect(Component)
  pageName="NotFound"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      404: Page Not Found
    </title>
    <meta
      content="This is the About..."
      name="description"
    />
  </HelmetWrapper>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          Whoops...
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          Something went wrong, please check your url and try again!
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Connect(Component)>
`;
